src/main/kotlin/org/opensearch/replication/action/replicationstatedetails/TransportUpdateReplicationStateDetails.kt

```diff
@@ -43,19 +44,24 @@ class TransportUpdateReplicationStateDetails @Inject constructor(transportServic
                 transportService, clusterService, threadPool, actionFilters, ::UpdateReplicationStateDetailsRequest, indexNameExpressionResolver),
         CoroutineScope by GlobalScope {
 
+    companion object {
+        const val DEFAULT_TIMEOUT = 60000L // Twice the default cluster publish timeout
+        const val DEFAULT_TIMEOUT_FACTOR = 3L // Thrice the current updated settings
+    }
```

Comment:
1. Wouldn't we end up always overriding with the default `cluster.publish.timeout` value of 30 secs since we are reading from settings?
2. Can we add reasoning of the timeout factor to be thrice? - Why does coroutine need to run longer?